### ISSUE TITLE
Form validation errors for camelCased entity properties are not mapped to the snake_cased form fields

### ISSUE THREAD

**Issue by nicocin** [symfony/symfony#22173]
| Q                | A
| ---------------- | -----
| Bug report?      | no
| Feature request? | yes
| BC Break report? | no
| RFC?             | no
| Symfony version  | 3.2.1

I use a form field using snake_case, while I use camelCase for the property name (and for the getter and setter method for this property) in the model.

When the form is submitted, the data is mapped to the model properties:
the PropertyPathMapper uses the PropertyAccessor to set the value. The PropertyAccessor uses a camelize() method to find getters and setters for the property.

But when a validation error occurs on such a property, it is not mapped back to the form field, because the ViolationMapper compares the camelCased propertyPath of the model property with the snake_cased propertyPath of the form field.

This may not be a bug, and people should not use a form field with another name than the property, (when not explicitly using the property_path option of the form field).
But I find it confusing that the mapping used for changing the model based on the form data works, but mapping the validation errors back to the form field does not.

**Comment by HeahDude** [comment]
Hello @nicocin, this looks like a bug to me, and I think it worked before, meanwhile you can explicit the property path thanks to the `property_path` option of the field:
```php
$builder->add('my_property', MyType::class, [
    'property_path' => 'myProperty',
];
```
that should do the trick.

**Comment by mario-seidel** [comment]
@nicocin I tried to reproduce that error in symfony v3.2.7 by creating a simple form and render it with a validation error. The model seems to be filled with the data entered correctly.
```
/**
     * @Route("/task", name="task_create")
     */
    public function createAction(Request $request)
    {
        $task = new Task();

        /** @var Form $form */
        $form = $this->createFormBuilder($task)
            ->add('task_name', TextType::class)
            ->add('dueDate', DateType::class)
            ->add('save', SubmitType::class, array('label' => 'Create Task'))
            ->getForm();

        $form->handleRequest($request);

        if ($form->isSubmitted() && $form->isValid()) {
            $task = $form->getData();
            return $this->redirectToRoute('homepage');
        }

        return $this->render('default/index.html.twig', [
            'form' => $form->createView(),
        ]);
    }
```
Can you confirm that this error also happens in latest version or provide some informations how to reproduce it?

**Comment by xabbuh** [comment]
@nicocin Can you have a look at @mario-seidel's comment?

**Comment by nicocin** [comment]
@mario-seidel, Thank you for trying to reproduce the problem.
The problem is not that data is not persisted, but that validation errors are not mapped to the form field.
I cloned the symfony-demo to show how to reproduce the issue:

1:
$ git clone git@github.com:nicocin/symfony-demo.git
$ cd symfony-demo
$ git checkout HEAD~2 
$ composer install --no-interaction
$ bin/console d:s:u --force
$ bin/console server:start

2: Go to http://127.0.0.1:8000/en/admin/post/  and sign in
3. Click create a new post
4. Do **not** fill in any of the fields and submit the form.
5. You will see validation errors for the first four fields, including Sub-title
6. Fill in each field, and submit. Click edit for the new post. You will see that the data, including the Sub-title was saved

7:
$ git checkout origin/HEAD~1
8. again, try to submit the 'new post' form without entering any data. You will see no validation error for the Sub-title
9. fill in all fields, except Sub-title. The form will reload, but without visible validation errors

10:
$ git checkout origin/HEAD
11. again, try to submit the 'new post' form without entering any data. You will see a validation error for the Sub-title field, but the error is not shown for the specific field, but for the form.


The three commits I made after cloning are very small, and should give an understanding of the problem, I hope.
